applications/ShallowWaterApplication/python_scripts/shallow_water_base_solver.py

```diff
@@ -134,55 +155,166 @@ def Initialize(self):
 
         print ("Mesh stage solver initialization finished")
 
-    def Solve(self):
-        # Solve equations on mesh
-        (self.solver).Solve()
+    def AdvanceInTime(self, current_time):
+        dt = self._ComputeDeltaTime()
+        new_time = current_time + dt
+
+        self.main_model_part.CloneTimeStep(new_time)
+        self.main_model_part.ProcessInfo[KratosMultiphysics.STEP] += 1
+
+        return new_time
+
+    def InitializeSolutionStep(self):
+        if self._TimeBufferIsInitialized():
+            self.solver.InitializeSolutionStep()
+
+    def Predict(self):
+        if self._TimeBufferIsInitialized():
+            self.solver.Predict()
+
+    def SolveSolutionStep(self):
+        if self._TimeBufferIsInitialized():
+            is_converged = self.solver.SolveSolutionStep()
+            return is_converged
+
+    def FinalizeSolutionStep(self):
+        if self._TimeBufferIsInitialized():
+            self.solver.FinalizeSolutionStep()
+
+    def Check(self):
+        self.solver.Check()
 
     def Clear(self):
-        (self.solver).Clear()
+        self.solver.Clear()
+
+    def Solve(self):
+        message  = "Calling ShallowWaterBaseSolver.Solve() method, which is deprecated\n",
+        message += "Please call the individual methods instead:\n",
+        message += "solver.InitializeSolutionStep()\n",
+        message += "solver.Predict()\n",
+        message += "solver.SolveSolutionStep()\n",
+        message += "solver.FinalizeSolutionStep()\n"
+
+        KratosMultiphysics.Logger.PrintWarning("ShallowWaterBaseSolver", message)
+
+        self.InitializeSolutionStep()
+        self.Predict()
+        self.SolveSolutionStep()
+        self.FinalizeSolutionStep()
+
+    #### Specific internal functions ####
 
-    def ComputeDeltaTime(self):
+    def _IsPrintingRank(self):
+        return self._is_printing_rank
+
+    def _TimeBufferIsInitialized(self):
+        # We always have one extra old step (step 0, read from input)
+        return self.main_model_part.ProcessInfo[KratosMultiphysics.STEP] + 1 >= self.GetMinimumBufferSize()
+
+    def _ComputeDeltaTime(self):
         # Automatic time step computation according to user defined CFL number
         if (self.settings["time_stepping"]["automatic_time_step"].GetBool()):
             delta_time = self.EstimateDeltaTimeUtility.EstimateDt()
         # User-defined delta time
         else:
             delta_time = self.settings["time_stepping"]["time_step"].GetDouble()
-        # Move particles utility needs to access delta_time
+        
+        # Move particles utility needs to read delta_time
+        self.main_model_part.ProcessInfo.SetValue(KratosMultiphysics.DELTA_TIME, delta_time)
```

Comment:
isnt this computed & set by CloneTimeStep?

Reply:
Yep

Reply:
I think CloneTimeStep is setting the new time

https://github.com/KratosMultiphysics/Kratos/blob/b8b4063c7d297747dd35c774b9d727650bfdd1cf/kratos/sources/model_part.cpp#L165

I am just updating the delta time variable